types/v1/metrics.go

```diff
@@ -30,25 +30,74 @@ type VolumeMetrics struct {
-	TotalWriteTime       json.Number `json:"TotalWriteTime"`
+	// TotalReadBytes is the total size of the read io's in byte.
+	TotalReadBytes json.Number `json:"TotalReadBytes"`
+	// Writes is the total no of write io's that's been
+	// written to the volume.
+	Writes json.Number `json:"WriteIOPS"`
+	// TotalWriteTime is the total time taken to write to
+	// the volume.
+	TotalWriteTime json.Number `json:"TotalWriteTime"`
+	// TotalWriteBlockCount is total no of block that has been
+	// written to the volume.
 	TotalWriteBlockCount json.Number `json:"TotalWriteBlockCount"`
-	TotalWriteBytes      json.Number `json:"TotalWriteBytes"`
-
+	// TotalWriteBytes is the total size of the write io's
+	// in byte.
+	TotalWriteBytes json.Number `json:"TotalWriteBytes"`
+	// UsedLogicalBlocks is the no of logical blocks that is
+	// used by volume.
 	UsedLogicalBlocks json.Number `json:"UsedLogicalBlocks"`
-	UsedBlocks        json.Number `json:"UsedBlocks"`
-	SectorSize        json.Number `json:"SectorSize"`
-	Size              json.Number `json:"Size"`
-	RevisionCounter   json.Number `json:"RevisionCounter"`
-	ReplicaCounter    json.Number `json:"ReplicaCounter"`
-	UpTime            json.Number `json:"UpTime"`
-	Name              string      `json:"Name"`
+	// UsedBlocks is the no of physical blocks used by volume.
+	// (each block is a physical sector)
+	UsedBlocks json.Number `json:"UsedBlocks"`
+	// SectorSize minimum storage unit of a hard drive.
+	// Most disk partitioning schemes are designed to have
+	// files occupy an integral number of sectors regardless
+	// of the file's actual size. Files that do not fill a whole
+	// sector will have the remainder of their last sector filled
+	// with zeroes. In practice, operating systems typically
+	// operate on blocks of data, which may span multiple sectors.
+	// (source: https://en.wikipedia.org/wiki/Disk_sector)
+	SectorSize json.Number `json:"SectorSize"`
+	// Size is the size of the volume created.
+	Size json.Number `json:"Size"`
+	// RevisionCounter is the no of times io's have been done
+	// on volume.
+	RevisionCounter json.Number `json:"RevisionCounter"`
+	// ReplicaCounter is the no of replicas connected to target
+	// or controller (istgt or jiva controller)
+	ReplicaCounter json.Number `json:"ReplicaCounter"`
+	// Uptime is the time since target is up.
+	UpTime json.Number `json:"UpTime"`
+	// Name is the name of the volume given while creation
+	Name string `json:"Name"`
+	// Replicas keeps the details about the replicas connected
+	// to the target.
+	Replicas []Replica `json:"Replicas"`
+	// Target status is the status of the target (RW/RO)
+	TargetStatus string `json:"Status"`
+}
+
+// Replica is used to store the info about the replicas
+// connected to the target.
+type Replica struct {
+	// Address is the address of the replica
+	Address string
+	// Mode is the mode of replica. (RW/RO for jiva and HEALTHY/
+	// DEGRADED for istgt)
+	Mode string
```

Comment:
Can we use a custom type instead of string?
Do we not require a json hint e.g. `json:"mode"`

Reply:
sure.